File: java/two-fer/src/main/java/Twofer.java
public class Twofer {
    public String twofer(String name) {
        String whom = isBlank(name) ? "you" : name;
        return "One for " + whom + ", one for me.";
    }

    private static boolean isBlank(String string) {
        return string == null || "".equals(string);
    }
}

File: go/two-fer/two_fer.go
// Package twofer provides a reply that shares between another person.
package twofer

import (
	"fmt"
)

// ShareWith returns a reply that shares something with another person.
func ShareWith(name string) string {
	if name == "" {
		name = "you"
	}
	return fmt.Sprintf("One for %s, one for me.", name)
}
